service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ProxyOptions.java

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.azure.sdk.iot.service;
+
+import lombok.Getter;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+
+/**
+ * The settings supported by this SDK when communicating to IoT Hub through a proxy. HTTP proxies are supported by this SDK,
+ * but only if the proxy does not require authentication.
+ */
+public class ProxyOptions
+{
+    @Getter
+    private Proxy proxy;
+
+    /**
+     * Create the proxy options for connecting to a proxy
+     * @param proxy the proxy to communicate through
+     */
+    public ProxyOptions(Proxy proxy)
+    {
+        if (proxy == null)
+        {
+            throw new IllegalArgumentException("Proxy cannot be null");
+        }
+
+        if (proxy.type() != Proxy.Type.HTTP)
```

Comment:
Q - this is for proxy support over Amqp_ws, then why don't we use socks proxy?

Reply:
The customer who prompted all this only needs support for HTTP proxies. I don't know offhand if amqps or amqps_ws work with socks proxies, or if proton-j has support for them

Reply:
I see, what does our device client currently use?

Reply:
HTTP proxies work for device clients using MQTT_WS, AMQPS_WS, and HTTP. In theory, SOCKs proxies work for device clients using MQTT